### DISCUSSION TITLE
bug:

### DISCUSSION THREAD

**Discussion by D3v01dZA** [LazyVim/LazyVim#5002] [Q&A] [answered]
### Did you check docs and existing issues?

- [x] I have read all the LazyVim docs
- [x] I have updated the plugin to the latest version before submitting this issue
- [x] I have searched the existing issues of LazyVim
- [x] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

NVIM v0.10.2

### Operating system/version

MacOS 15.1.1

### Describe the bug

As a newer user I rely on WhichKey. When in a buffer that has both an lsp and a git root, WhichKey only shows me git keys when using leader g, while invoking :WhichKey g (:WhichKey and then pressing g) shows me only lsp keys. In both cases I would expect both git and lsp keys to be shown at the same time. 

As a side not the git bindings conflict with http://www.lazyvim.org/extras/coding/mini-surround and I would prefer to have that over git but can't find how the git commands are even making it in except by defaults.

### Steps To Reproduce

1. Install lazy
2. Open a file that's in a git repo and will attach an LSP
3. Use leader g
4. See only git commands show up
5. Do :WhichKey g
6. See only lspcommands show up

### Expected Behavior

I expect which key to show me both the git keys and the lsp keys available to me, not just git. In my opinion if I had to choose one or the other I would want the lsp keys shown.

### Repro

```lua
vim.env.LAZY_STDPATH = ".repro"
load(vim.fn.system("curl -s https://raw.githubusercontent.com/folke/lazy.nvim/main/bootstrap.lua"))()

require("lazy.minit").repro({
  spec = {
    { "LazyVim/LazyVim", import = "lazyvim.plugins" },
    -- add any other plugins here
  },
})
```

**Comment by folke** [comment, marked as answer]
Please use discussions next time for user questions as this is **NOT** a bug.

There's a difference between `<leader>g` mappings and `g` mappings.
That's not the same.

**Comment by D3v01dZA** [comment]
First off, my apologies, somehow last night I was convinced I had found a bug, I will do better next time.

You are of course right that leader g and g are different and as a bonus which key does indeed have all mappings available in the right place including surround.

Thank you!
